### PR TITLE
Bookstore: Function add reducers and actions

### PR DESCRIPTION
**Pull request by gitbygaurav**
### Hey CR 😊

AidanHelloAidanGallagherHelloGIF

In this PR, I have :

- Added Redux Toolkit.
- Structured application files using a "feature folder" approach and used the [ducks pattern](https://github.com/erikras/ducks-modular-redux) for Redux files.
- Written book's actions and reducer (in /src/redux/books/books.js) using the [ducks pattern](https://github.com/erikras/ducks-modular-redux).
  - Defined action types for adding and removing a book.
  - Set the initial state to be an empty array of books.
  - Exported Action Creators for actions.
  - Written reducer and exported it as default.
     - Defined state changes for the actions that were created.
     - In case of unknown action - returned current state.
- Written categories actions and reducer (in /src/redux/categories/categories.js) using the [ducks pattern](https://github.com/erikras/ducks-modular-redux).
  - Defined an action type for checking the status.
  - Set the initial state to be an empty array of categories.
  - Exported Action Creators for the actions.   
  - Written reducer and exported it as default. 
     - For the check status action returned string "Under construction";
     - In case of unknown action - returned the current state.
- Configured the Redux Store (/src/redux/configureStore.js): 
   - Imported the necessary methods from Redux Toolkit.
   - Combined both reducers into a root reducer by using configureStore function.

Thanks 🍻
Gaurav

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
-  </BrowserRouter>
+  <Provider store={booksStore}>
+    <BrowserRouter>
+      <Navbar />
+      <Routes>
+        <Route path="/" element={<BooksPage />} />
+        <Route path="/categories" element={<CategoriesPage />} />
+      </Routes>
+    </BrowserRouter>
+  </Provider>
```
Comment (Lameck1) [inline src/App.js:18]: This is a test inline comment.

**Review by Lameck1 — COMMENTED** [review]
Hello @gauravgangwarweb,

This is a test code review trying to investigate some issues. Please ignore it.

File: src/redux/books/books.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+const ADD_BOOK = 'ADD_BOOK';
+const REMOVE_BOOK = 'REMOVE_BOOK';
```
Comment (rloterh) [inline src/redux/books/books.js]: - [ ] With reference to the _duck pattern_, assigning values to _ADD_BOOK_ and _REMOVE_BOOK_ action types should be similar to:
```javascript
       const ADD_BOOK = bookstore/books/ADD_BOOK;
```

and

```javascript
      const REMOVE_BOOK = bookstore/books/REMOVE_BOOK;
```

Kindly ensure to improve upon these aspects.

File: src/redux/categories/categories.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+const CHECK_STATUS = 'CHECK_STATUS';
```
Comment (rloterh) [inline src/redux/categories/categories.js]: - [ ] Similarly, assigning value to _CHECK_STATUS_ action type with respect to the _duck patterns approach_ should be something like below:
```Javascript
      const CHECK_STATUS = bookstore/categories/CHECK_STATUS;
```
 
Kindly ensure to improve upon this aspect.

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Gaurav, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project, but you are almost there! 🚦 🚦  🏃‍♂️
 
## Highlights 👍  
- Nice job so far!
- Professional ReadMe!

## Required Changes ♻️
_Kindly check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by gitbygaurav** [comment]
Hii @rloterh 👋 
Thanks for reviewing my PR and for suggesting the changes. I will make the changes as required.

**Review by DammyShittu — APPROVED** [review]
Hi @gauravgangwarweb,

## APPROVED 🟢

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by gitbygaurav** [comment]
Hii @DammyShittu 👋

ThankYouStickerThanksStickerGIF



Thanks for reviewing and approving my PR.

**Pull request merged**